mgmt/rest/v2/plugin.go

```diff
@@ -183,40 +211,44 @@ func (s *apiV2) loadPlugin(w http.ResponseWriter, r *http.Request, _ httprouter.
 				}
 			}
 
-			// A little sanity checking for files being passed into the API server.
-			// First file passed in should be the plugin. If the first file is a signature
-			// file, an error is returned. The signature file should be the second
-			// file passed to the API server. If the second file does not have the ".asc"
-			// extension, an error is returned.
-			// If we loop around more than twice before receiving io.EOF, then
-			// an error is returned.
-
-			switch {
-			case i == 0:
-				if filepath.Ext(p.FileName()) == ".asc" {
-					e := errors.New("Error: first file passed to load plugin api can not be signature file")
-					Write(400, FromError(e), w)
+			switch p.FormName() {
+			case "ca_certs":
+				fn, err := createTempFile(b, TLSCACertsPrefix)
```

Comment:
@iza, thanks for your review. The situation here is different than v1. V1's client always resides inside Snap on the same machine. V2 passes the content of certs and key over to Snap as `snap-cli` may install on a different server. By leveraging what is available in Snap (requires paths of certs and the key), that's how creating temp files are necessary. Otherwise, I need to change Snap's code to use the content of certs and the key directly. 

Agreed, I don't prefer to use temp files. Unless we assume that those certs and the key files are always accessible even when they're on a different server. 

Please let me know how you like to proceed.